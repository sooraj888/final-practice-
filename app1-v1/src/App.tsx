import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Box } from "@material-ui/core";

function App() {
  return <Box className="App"></Box>;
}

export default App;
