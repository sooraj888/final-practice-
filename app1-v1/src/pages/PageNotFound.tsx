import { Box } from "@material-ui/core";
import React from "react";

const PageNotFound = (): JSX.Element => {
  return (
    <Box>
      <h1>Page Not Found</h1>
    </Box>
  );
};

export default PageNotFound;
