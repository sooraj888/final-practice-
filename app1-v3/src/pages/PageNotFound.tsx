import React from "react";

const PageNotFound = (): JSX.Element => {
  return <div>Page Not Found</div>;
};

export default PageNotFound;
