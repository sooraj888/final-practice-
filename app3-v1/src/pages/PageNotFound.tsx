import { Button } from "@mui/material";
import React from "react";
import { Link } from "react-router-dom";

const PageNotFound = (): JSX.Element => {
  return (
    <div>
      <h1>Page Not Found </h1>
      <Link to="/">
        <Button>Back</Button>
      </Link>
    </div>
  );
};

export default PageNotFound;
