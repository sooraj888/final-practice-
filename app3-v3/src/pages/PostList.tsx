import {
  Box,
  Button,
  CircularProgress,
  Pagination,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from "@mui/material";
import React from "react";
import { NavigateFunction, useNavigate } from "react-router-dom";
import NavBar from "../Componets/NavBar";
import ShowError from "./../utility/ShowError";

type PostListType = {
  page: number;
  data: any[];
  setPage: React.Dispatch<React.SetStateAction<number>>;
  error: boolean;
};

const PostList = ({ page, data, setPage, error }: PostListType) => {
  const navigate: NavigateFunction = useNavigate();
  const handldeOnClickSelect = (item: string): void => {
    navigate("/rawJson", { state: item });
  };
  const handlePagination = (e: any, value: number): void => {
    setPage(value);
  };

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-around",
        alignItems: "center",
      }}
    >
      {error ? (
        <ShowError></ShowError>
      ) : data.length === 0 ? (
        <>
          <CircularProgress></CircularProgress>
          <p>Loading...</p>
        </>
      ) : (
        <Box
          sx={{
            dispaly: "flex",
            justifyContent: "center",
            alignItems: "center",
            textAlign: "center",
            flexDirection: "column",
          }}
        >
          <NavBar />
          <TableContainer sx={{ maxHeight: 440 }}>
            <Table stickyHeader>
              <TableHead>
                <TableRow>
                  <TableCell>Created-At</TableCell>
                  <TableCell>Title</TableCell>
                  <TableCell>URL</TableCell>
                  <TableCell>Author</TableCell>
                  <TableCell>Select</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {data.length > 0 &&
                  data[page - 1].map((item: any, key: number): JSX.Element => {
                    return (
                      <TableRow key={key}>
                        <TableCell>{item?.created_at}</TableCell>
                        <TableCell>{item?.title}</TableCell>
                        <TableCell>{item?.url}</TableCell>
                        <TableCell>{item?.author}</TableCell>
                        <TableCell>
                          <Button onClick={() => handldeOnClickSelect(item)}>
                            select
                          </Button>
                        </TableCell>
                      </TableRow>
                    );
                  })}
              </TableBody>
            </Table>
          </TableContainer>
        </Box>
      )}
      {!error && data.length !== 0 && (
        <Pagination
          page={page}
          count={data.length}
          onChange={handlePagination}
        ></Pagination>
      )}
    </Box>
  );
};

export default PostList;
