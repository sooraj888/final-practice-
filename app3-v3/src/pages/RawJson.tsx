import React from "react";
import { useLocation } from "react-router-dom";

const RawJson = (): JSX.Element => {
  const localState: unknown = useLocation().state;

  if (!localState) {
    return <h1>Data notFound</h1>;
  }
  return (
    <div>
      <h1>Raw JSON</h1>
      {JSON.stringify(localState)}
    </div>
  );
};

export default RawJson;
