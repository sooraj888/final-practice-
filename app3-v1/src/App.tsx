import React, { useEffect, useState } from "react";
import logo from "./logo.svg";
import "./App.css";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import NavBar from "./components/NavBar";
import PostListPage from "./pages/PostListPage";
import RawJson from "./pages/RawJson";
import PageNotFound from "./pages/PageNotFound";
import axios from "axios";

export type postDataType = Array<Array<any>>;

function App(): JSX.Element {
  const [postData, setPostData] = useState<postDataType>([]);

  const [page, setPage] = useState<number>(1);

  let count: number = 0;

  let myInterval: NodeJS.Timer;

  const fetchData = async (): Promise<any> => {
    const responce: any = await axios.get(
      `http://hn.algolia.com/api/v1/search?query=bar&page=${count}`
    );
    console.log(responce);
    if (!responce?.data?.exhaustiveNbHits) {
      setPostData((prev: postDataType): postDataType => {
        let updatedArr: postDataType = [...prev];
        updatedArr = [...updatedArr, responce?.data?.hits];
        return updatedArr;
      });
      count += 1;
    } else {
      clearInterval(myInterval);
    }
  };

  useEffect((): void => {
    fetchData();
    myInterval = setInterval(fetchData, 3000);
  }, []);

  useEffect((): void => {
    console.log(postData);
  }, [postData]);

  return (
    <div className="App">
      <NavBar />
      <BrowserRouter>
        <Routes>
          <Route
            path="/"
            element={
              <PostListPage
                postData={postData}
                page={page}
                setPage={setPage}
              ></PostListPage>
            }
          />
          <Route path="/rawJson" element={<RawJson />}></Route>
          <Route path="*" element={<PageNotFound />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
