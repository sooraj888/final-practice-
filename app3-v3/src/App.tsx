import React, { useEffect, useState } from "react";
import logo from "./logo.svg";
import "./App.css";
import axios, { AxiosResponse } from "axios";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import PostList from "./pages/PostList";
import RawJson from "./pages/RawJson";
import PageNotFound from "./pages/PageNotFound";
import NavBar from "./Componets/NavBar";

function App(): JSX.Element {
  const [data, setData] = useState<any[]>([]);
  let count: number = 0;
  const [page, setPage] = useState<number>(1);
  let intervel: NodeJS.Timer;

  const [error, setError] = useState<boolean>(false);
  const url: string = "https://hn.algolia.com/api/v1/search?query=bar&page=";

  const fetchData = async (): Promise<any> => {
    try {
      setError(false);
      const responce: AxiosResponse<any, any> = await axios.get(url + count);
      if (!responce?.data?.exhaustiveNbHits) {
        setData((prev): any[] => [...prev, responce?.data?.hits]);
        count += 1;
      } else {
        clearInterval(intervel);
      }
    } catch (e) {
      setError(true);
    } finally {
    }
  };

  useEffect((): void => {
    fetchData();
    intervel = setInterval(fetchData, 10000);
  }, []);
  useEffect((): void => {
    console.log(data);
  }, [data]);

  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route
            path="/"
            element={
              <PostList
                setPage={setPage}
                data={data}
                page={page}
                error={error}
              />
            }
          />
          <Route path="/rawJson" element={<RawJson />} />
          <Route path="*" element={<PageNotFound />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;

//         ` http://hn.algolia.com/api/v1/search?query=bar&page=${count}
