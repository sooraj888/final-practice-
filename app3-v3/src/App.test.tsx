import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import PageNotFound from "./pages/PageNotFound";
import RawJson from "./pages/RawJson";
import { BrowserRouter } from "react-router-dom";
import PostList from "./pages/PostList";

describe("test on PageNotFound components ", () => {
  test("rendering PageNotFound", () => {
    render(<PageNotFound />);
    screen.debug();
  });
});

describe("test on PageNotFound components ", () => {
  test("finding test on  PageNotFound", () => {
    render(<PageNotFound />);
    const ele = screen.getByText(/Page not found/);
    expect(ele).toBeInTheDocument();
  });
});

describe("test on PageNotFound components ", () => {
  test("snapshot test test on  PageNotFound", () => {
    const { container } = render(<PageNotFound />);
    expect(container).toMatchSnapshot();
  });
});

describe("test on RawJson components ", () => {
  test("rendering RawJson", () => {
    render(
      <BrowserRouter>
        <RawJson />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on RawJson components ", () => {
  test("finding test on  RawJson", () => {
    render(
      <BrowserRouter>
        <RawJson />
      </BrowserRouter>
    );
    const ele = screen.getByText(/Data notFound/);
    expect(ele).toBeInTheDocument();
  });
});

describe("test on RawJson components ", () => {
  test("snapshot test test on  RawJson", () => {
    const { container } = render(
      <BrowserRouter>
        <RawJson />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("test on PostList components ", () => {
  test("rendering PostList", () => {
    render(
      <BrowserRouter>
        <PostList error={false} page={1} data={[]} setPage={jest.fn()} />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on PostList components ", () => {
  test("rendering PostList", () => {
    render(
      <BrowserRouter>
        <PostList error={false} page={1} data={[]} setPage={jest.fn()} />
      </BrowserRouter>
    );
    expect(screen.getByText(/Loading.../)).toBeInTheDocument();
  });
});

const hits = [
  [
    {
      created_at: "123",
      title: "sooraj1",
      ur: "http:/sooraj.com",
      author: "soory",
    },
  ],
  [{ created_at: "124", title: "raj1", ur: "http:/raj.com", author: "sry" }],
];

const prop = { page: 1, data: hits, setPage: jest.fn() };

describe("test on PostList components ", () => {
  test("rendering PostList", () => {
    render(
      <BrowserRouter>
        <PostList error={false} page={1} data={hits} setPage={jest.fn()} />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on PostList components ", () => {
  test(" test test on  PostList 2", () => {
    const { container } = render(
      <BrowserRouter>
        <PostList error={false} page={1} data={hits} setPage={jest.fn()} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("test on PostList components ", () => {
  test(" test on  PostList", () => {
    const { container } = render(
      <BrowserRouter>
        <PostList error={false} {...prop} />
      </BrowserRouter>
    );

    fireEvent.click(screen.getAllByRole("button")[2]);
    expect(prop.setPage).toHaveBeenCalledTimes(1);
  });
});
