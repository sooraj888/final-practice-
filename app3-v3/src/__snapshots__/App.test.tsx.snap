// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on PageNotFound components  snapshot test test on  PageNotFound 1`] = `
<div>
  <h1>
    Page not found 
  </h1>
</div>
`;

exports[`test on PageNotFound components  snapshot test test on  RawJson 1`] = `
<div>
  <h1>
    Data notFound
  </h1>
</div>
`;

exports[`test on PostList components   test test on  PostList 2 1`] = `
<div>
  <div
    class="MuiBox-root css-560y87"
  >
    <div
      class="MuiBox-root css-1hnzlut"
    >
      <div
        class="MuiTableContainer-root css-41abqd-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Created-At
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Title
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                URL
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Author
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Select
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                123
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                sooraj1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                soory
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  select
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
    >
      <ul
        class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            2
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`test on PostList components  snapshot test test on  PostList 1`] = `
<div>
  <div
    class="MuiBox-root css-560y87"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <p>
      Loading...
    </p>
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
    >
      <ul
        class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`test on RawJson components  snapshot test test on  RawJson 1`] = `
<div>
  <h1>
    Data notFound
  </h1>
</div>
`;
